source/base/bounding_box.cc

```diff
@@ -0,0 +1,43 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by Giovanni Alzetta
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/bounding_box.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+template <int spacedim, typename Number>
+bool BoundingBox<spacedim,Number>::point_inside (const Point<spacedim, Number> &p) const
+{
+  for (unsigned int i=0; i < spacedim; ++i)
+    {
+      //The point is outside the box if it is greater (or smaller) than both coordinates
+      if ( (this->boundary_points.first[i] < p[i] && this->boundary_points.second[i] < p[i])
+           || (this->boundary_points.first[i] > p[i] && this->boundary_points.second[i] > p[i]))
```

Comment:
If you adopt the convention that the first point needs to be to the left of the right point, then you can avoid half of the comparisons here.